testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_custom_conflicts.py

```diff
@@ -0,0 +1,1013 @@
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    doc_ids = db.getDocIds(cbl_db)
+    sg_docs = sg_client.get_bulk_docs(sg_url, sg_db, doc_ids, session)[0]
+
+    # creating conflict for docs on SG
+    sg_client.delete_bulk_docs(url=sg_url, db=sg_db, docs=sg_docs, auth=session)
+    # creating conflict for docs on CBL
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+    for doc_id in cbl_docs:
+        for _ in range(2):
+            log_info("Updating CBL Doc - {}".format(doc_id))
+            data = cbl_docs[doc_id]
+            data = property_updater(data)
+            data["cbl_random"] = random_string(length=10, printable=True)
+            db.updateDocument(cbl_db, doc_id=doc_id, data=data)
+
+    repl_config = replicator.configure(cbl_db, sg_blip_url, continuous=True, channels=channels,
+                                       replicator_authenticator=replicator_authenticator,
+                                       replication_type=replicator_type, conflict_resolver="delete_not_win")
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    # printing doc content before replication conflicted docs
+    try:
+        sg_docs_content = sg_client.get_bulk_docs(sg_url, sg_db, doc_ids, session)[0]
+    except Exception as err:
+        log_info("Requesting for deleted docs from SG: {}".format(err))
+        sg_docs_content = []
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+    if replicator_type == "pull":
+        assert sg_docs_content == [], "CCR failed to resolved conflict with delete_not_win and replicated content to SG"
+        assert cbl_docs, "CCR failed to resolved conflict with delete_not_win"
+    elif replicator_type == "push_pull":
+        for sg_doc in sg_docs_content:
+            doc_id = sg_doc["_id"]
+            cbl_doc = cbl_docs[doc_id]
+            assert sg_doc["sg_new_update1"] == cbl_doc["sg_new_update1"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert sg_doc["sg_new_update2"] == cbl_doc["sg_new_update2"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert sg_doc["sg_new_update3"] == cbl_doc["sg_new_update3"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert "random" not in sg_doc, "CCR failed to resolve conflict with delete_not_win"
+            assert "cbl_random" in sg_doc, "CCR failed to resolve conflict with delete_not_win"
+
```

Comment:
it doesn't matter, as the resolver in place is there to check that we are not allowing the delete functionality to effect the local copy. That's why I have written so many resolver so that each scenario is covered in isolation as user will probably have one resolver with all the complex logic in it. Other resolver does cover that scenario